include/kframework/lemmas/bitwise-simplification.k

```diff
@@ -43,4 +43,12 @@ module BITWISE-SIMPLIFICATION [symbolic]
     rule notMaxUInt160 &Int X => 0 requires #rangeUInt(160, X) [simplification]
     rule maxUInt160 &Int (X |Int (notMaxUInt160 &Int Y:Int)) => X requires #rangeUInt(160, X) andBool 0 <=Int Y [simplification]
 
+    rule [lengthBytes-upInt-32-lower-bound]:
+      lengthBytes(X) <=Int notMaxUInt5 &Int ( lengthBytes(X) +Int 31 ) => true
+      [simplification, smt-lemma]
+
+    rule [lengthBytes-upInt-32-upper-bound]:
+      lengthBytes(X) +Int 32 >Int notMaxUInt5 &Int ( lengthBytes(X) +Int 31 ) => true
+      [simplification, smt-lemma]
```

Comment:
Counterexample: `lengthBytes(X) >=Int 32`. This is because `notMaxUInt5` is `maxUInt256 xorInt maxUInt5`.

Reply:
I think just adding `requires lengthBytes(X) <=Int 32` should be fine?

The counterexample only applies to the first lemma I suppose.

Reply:
Yes, that will work.

Reply:
It is not clear to me that Optimism proofs will pass with this, because there is no built-in limit on the length of `Bytes`. There probably should be one, since `Bytes` of length `pow256` could never possibly appear?